synapse/lib/stormsvc.py

```diff
@@ -125,6 +130,22 @@ class StormSvcClient(s_base.Base, s_stormtypes.StormType):
         if 'StormSvc' in names:
 
             self.info = await proxy.getStormSvcInfo()
+            try:
+                init = self.info.get('init', '')
+                if init:
+                    await s_common.aspin(self.core.storm(init))
+
+            except asyncio.CancelledError:  # pragma: no cover
+                raise
+
+            except Exception:
+                logger.exception(f'init failed for service {self.name} ({self.iden})')
+
+            fini = self.info.get('fini', '')
+            if fini:
+                async def finifunc():
+                    await s_common.aspin(self.core.storm(fini))
+                self.onfini(finifunc)
```

Comment:
do these execute early enough during shutdown to not exception? ( and are they currently designed for "uninstall" or "cortex shutdown" or "service fell off line"?  Looks like "cortex shutdown" currently?

Reply:
Is the use case her for registration/deregistration, or startup/shutdown? Because encouraging storm code during shutdown is risky.  A malicious service could do a shutdown routine of `.created | sleep 1000` which could make the Cortex appear to hang during shutdown. 

In the shutdown case you have in a test, we're actually racing between "is the hive shutdown" or "is the stormsvc fini function" being called; because we're iterating over a Base.tofini weakref set.  There are other Cortex components which may also be shut down (like the Boss) before our stormsvc is done; and the Boss does some storm management and task cleanup during shutdown.